### PR TITLE
bpo-38956: don't print BooleanOptionalAction's default twice

### PR DESCRIPTION
**Pull request by mhils**
This PR is a less intrusive fix to not double-print the default value when using `argparse.BooleanOptionalAction` with  `ArgumentDefaultsHelpFormatter`. The original bug report ([bpo-38956](https://bugs.python.org/issue38956)) was about removing the default entirely, but @rhettinger made a sensible point in https://github.com/python/cpython/pull/17447#discussion_r429630944 that a default should always been shown here. While #17447 seems to work as intended, progress stalled a bit presumably because it's not entirely trivial to review. This PR proposes a more straightforward attempt to solve the problem, which can hopefully be reviewed more easily. If the maintainers ultimately prefers #17447 that is perfectly fine with me of course. 😃 

In short, in this PR we make use of the fact that `ArgumentDefaultsHelpFormatter` already checks if the default is already included:

https://github.com/python/cpython/blob/2b496e79293a8b80e8ba0e514e186b3b1467b64b/Lib/argparse.py#L693

This condition previously did not trigger because `BooleanOptionalAction` already did the string interpolation directly.

https://bugs.python.org/issue38956

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2021-08-09-13-17-10.bpo-38956.owWLNv.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+:class:`argparse.BooleanOptionalAction`'s default value is now longer printed twice when used with :class:`argparse.ArgumentDefaultsHelpFormatter`.
```
Comment (iamrajhans) [inline Misc/NEWS.d/next/Library/2021-08-09-13-17-10.bpo-38956.owWLNv.rst]: @mhils i was just going through the PR and found one typo, `value is now longer ....` it should be `value is no longer ....`

Comment (mhils) [reply Misc/NEWS.d/next/Library/2021-08-09-13-17-10.bpo-38956.owWLNv.rst]: Good catch, thanks! 🍰

**Comment by ambv** [comment]
Confirmed that the test fails without the source change:
```
-   --taz, --no-taz  Whether to taz it (default: True)
+   --taz, --no-taz  Whether to taz it (default: True) (default: True)
?                                                ++++++++++++++++
    --quux QUUX      Set the quux (default: 42)
```

(that diffing algorithm is weird)

**Comment by abadger** [comment]
Hi, I don't think this was the right fix for this issue (or maybe better stated as... this is an incomplete fix that leaves two diverging code paths which should be consolidated).  Please see https://bugs.python.org/issue44587 for my explanation of what I think should be done (Option 2 if we want to go the nonintrusive route).  I have an implementation of option 2 vendored into one of my projects here: https://github.com/ansible-community/antsibull/commit/67f6d1a2c091f6e0d9733f2f1212f86b918ed6b3#diff-f515d1b9734c55a77e0a94b093b41eb1102a986be60631df39ac1513895aba07  I can pull that out into a pr but I had been waiting on someone to confirm that Option 2 is really the desired outcome considering the architecture of argparse.

**Comment by mhils** [comment]
Thanks @ambv! ❤️ 

@abadger: My intention with this PR was to unblock a somewhat stalled issue with changes that are trivial to review. bpo-44587 wasn't linked at bpo-38956 so I wasn't aware of it. The changes here shouldn't affect the usefulness of what you proposed, I'd suggest you go ahead and open a PR for your option 2. :)

**Comment by abadger** [comment]
<nod> i wanted to point it out because this change doesn't fix the rest of
the problems caused by the root issue. For instance, as reported in my
issue, SUPPRESS will still be displayed as a default value.

**Comment by ambv** [comment]
@abadger, this was merged because it's a rather trivial improvement over the status quo. I intend to also release this fix in 3.9.7 on August 30th. Your idea in BPO-44587 to further improve this while retaining displaying the default value is interesting. If you prepare a PR we can work with, we'll look into merging that as well.
